File: examples/tutorial/c/tut_allocator.c
//////////////////////////////////////////////////////////////////////////////
// Copyright (c) 2016-23, Lawrence Livermore National Security, LLC and Umpire
// project contributors. See the COPYRIGHT file for details.
//
// SPDX-License-Identifier: (MIT)
//////////////////////////////////////////////////////////////////////////////
#include <stdio.h>

#include "umpire/interface/c_fortran/umpire.h"

#define SIZE 1024

int main() {
  /* _sphinx_tag_tut_c_get_allocator_start */
  umpire_resourcemanager rm;
  umpire_resourcemanager_get_instance(&rm);

  umpire_allocator allocator;
  umpire_resourcemanager_get_allocator_by_name(&rm, "HOST", &allocator);
  /* _sphinx_tag_tut_c_get_allocator_end */

  /* _sphinx_tag_tut_c_allocate_start */
  double* data = (double*) umpire_allocator_allocate(&allocator, SIZE*sizeof(double));

  printf("Allocated %lu bytes using the %s allocator...", (SIZE*sizeof(double)), umpire_allocator_get_name(&allocator));

  umpire_allocator_deallocate(&allocator, data);
  /* _sphinx_tag_tut_c_allocate_end */

  printf("deallocated.\n");

  return 0;
}

File: examples/tutorial/fortran/tut_allocator.f
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
! Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
! project contributors. See the COPYRIGHT file for details.
!
! SPDX-License-Identifier: (MIT)
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
    
program fortran_test
      use umpire_mod
      implicit none
      logical ok

      integer(C_INT), pointer :: array(:)
      type(UmpireAllocator) allocator
      type(UmpireResourceManager) rm

      ! _umpire_tut_get_allocator_start
      rm = rm%get_instance()
      allocator = rm%get_allocator_by_id(0)
      ! _umpire_tut_get_allocator_end

      ! _umpire_tut_allocate_start
      call allocator%allocate(array, [ 10 ])

      write(10,*) "Allocated array of size ", 10

      call allocator%deallocate(array)
      ! _umpire_tut_allocate_end

      write(10,*) "deallocated."
end program fortran_test
